woodwork/type_sys/type_system.py

```diff
@@ -310,6 +310,10 @@ def get_random_sample(series_, **kwargs):
             series = get_random_sample(series, **kw_args_sampling)
         else:
             # Dask and Spark don't accept the n argument
+
+            # prevent division by zero error
+            if not len(series):
```

Comment:
I considered moving this before the `if` block on L305. I am open to moving it there, but since we don't divide by `len` in L305-L310, I think it should be fine to keep it here too.

Reply:
I would leave it here - you don't need it for the pandas flow, so we shouldn't do the work of calculating the len until we need it.

Also, I would store the len in a variable and then use that variable on L315 and L317 so you don't end up calculating the length twice for cases where there are values present.

Reply:
I could be wrong, but I remember reading that `len` is a `O(1)` attribute lookup in Python.

Reply:
I agree keeping it on L315 makes sense though, seeing as we don't use the `frac` argument for Pandas.

Reply:
> I could be wrong, but I remember reading that `len` is a `O(1)` attribute lookup in Python.

I'm not exactly sure on that, but in this code block we are dealing with Dask and Spark dataframes which can be quite large. My concern is that we will end up having to read in those dataframes into memory twice to get the len. I'm not sure how those libraries implement len, but I think it is potentially costly.

Reply:
Ah, got it. That makes sense! I looked at the code again, and there's no way we hit L316 and L317 at the same time, but we can cache and make L317 and L320 more efficient!

Reply:
That makes sense! I cached on L320 too to make it more efficient as well :)